platforms/database.go

```diff
@@ -0,0 +1,93 @@
+package platforms
+
+import (
+	"runtime"
+	"strings"
+)
+
+// validComboString defines valid GOOS/GOARCH combinations.
+// Should be updated from https://golang.org/doc/install/source#environment.
+// This is why we leave it as a string.
+const (
+	validComboString = `
+$GOOS	$GOARCH
+android	arm
+darwin	386
+darwin	amd64
+darwin	arm
+darwin	arm64
+dragonfly	amd64
+freebsd	386
+freebsd	amd64
+freebsd	arm
+linux	386
+linux	amd64
+linux	arm
+linux	arm64
+linux	ppc64
+linux	ppc64le
+linux	mips
+linux	mipsle
+linux	mips64
+linux	mips64le
```

Comment:
@tophj-ibm I've added a few test cases for `linux/s390x` to ensure its covered. I'm going to structure this database a little more carefully to avoid these kinds of problems. I don't want this package to become an impediment to wider platform support!